x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,166 @@
-            },
-          };
-        }),
-        mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
-      );
+
+      const queryFactory: TimelineFactory<T> = timelineFactory[factoryQueryType];
+
+      if (entityType != null && entityType === EntityType.ALERTS) {
+        return timelineAlertsSearchStrategy({
+          es: esAsInternal,
+          request,
+          options,
+          deps,
+          queryFactory,
+          alerting,
+          alertConsumers: alertConsumers ?? [],
+        });
+      } else {
+        return timelineSearchStrategy({ es, request, options, deps, queryFactory });
+      }
     },
     cancel: async (id, options, deps) => {
       if (es.cancel) {
         return es.cancel(id, options, deps);
+      } else if (esAsInternal.cancel) {
+        return esAsInternal.cancel(id, options, deps);
       }
     },
   };
 };
+
+const timelineSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  queryFactory: TimelineFactory<T>;
+}) => {
+  const dsl = queryFactory.buildDsl(request);
+  return es.search({ ...request, params: dsl }, options, deps).pipe(
+    map((response) => {
+      return {
+        ...response,
+        ...{
+          rawResponse: shimHitsTotal(response.rawResponse, options),
+        },
+      };
+    }),
+    mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
+  );
+};
+
+const timelineAlertsSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+  alerting,
+  alertConsumers,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  alerting: AlertPluginStartContract;
+  queryFactory: TimelineFactory<T>;
+  alertConsumers: ALERTS_CONSUMERS[];
+}) => {
+  const allFeatureIdsValid = alertConsumers.every((id) => isValidFeatureId(id));
+
+  if (!allFeatureIdsValid) {
+    throw new Error('An invalid alerts consumer feature id was provided');
+  }
+
+  const indices = alertConsumers.flatMap((consumer) => mapConsumerToIndexName[consumer]);
+  const requestWithAlertsIndices = { ...request, defaultIndex: indices, indexName: indices };
+
+  // Note: Alerts RBAC are built off of the alerting's authorization class, which
+  // is why we are pulling from alerting, not ther alertsClient here
+  const alertingAuthorizationClient = alerting.getAlertingAuthorizationWithRequest(deps.request);
+
+  const getAuthFilter = async () =>
+    alertingAuthorizationClient.getFindAuthorizationFilter(AlertingAuthorizationEntity.Alert, {
+      type: AlertingAuthorizationFilterType.ESDSL,
+      fieldNames: {
+        consumer: OWNER,
+        ruleTypeId: RULE_ID,
+        spaceIds: SPACE_IDS,
+      },
+      includeSpaceId: true,
```

Comment:
what happen if kibana is config without spaces?

Reply:
So I redid the tests completely to try to add more test cases. You'll see that I moved the auth specific tests for timeline under a new `timeline` folder in api integration tests. It includes a suite for when its `security_and_spaces`, `spaces_only` and `security_only`. There's two tests that aren't passing right now I'm working on, but this should give us better coverage.

@dhurley14 do we have test cases for the endpoints to deal with when spaces is disabled? These tests only hit the code path where we hit `getFindAuthorizationFilter`.

Reply:
hmmm I do not believe we do. We will need a set of integration tests with a configuration where the spaces plugin is disabled. It will be similar to the `spaces_only` dir but just `security_only` or something like that. This is a good point.